pkg/util/p2_exec.go

```diff
@@ -0,0 +1,41 @@
+package util
+
+type P2Exec struct {
+	Bin string
+}
+
+// Struct that can be used to build an invocation of p2-exec with any
+// applicable flags
+type P2ExecArgs struct {
+	User             string
+	EnvDir           string
+	NoLimits         bool
+	CgroupName       string
+	CgroupConfigName string
+	Command          string
+}
+
+func (p P2Exec) BuildWithArgs(args P2ExecArgs) []string {
```

Comment:
nit: Can you shorten this to `Build()`?  I think the `ActionWithThing1(t)` naming pattern is more appropriate when you're trying to distinguish it from a default action (e.g., `Action()`) or different different styles of arguments (e.g., `ActionWithThing2(t)`).  That doesn't apply here.

Or, name the method for what it produces: `CommandLine()`